include/fiction/algorithms/simulation/sidb/operational_domain.hpp

```diff
@@ -0,0 +1,1260 @@
+        {
+            op_domain.operational_values[param_point] = operational_domain::operational_status::OPERATIONAL;
+
+            return operational_domain::operational_status::OPERATIONAL;
+        };
+
+        const auto non_operational = [this, &param_point]()
+        {
+            op_domain.operational_values[param_point] = operational_domain::operational_status::NON_OPERATIONAL;
+
+            return operational_domain::operational_status::NON_OPERATIONAL;
+        };
+
+        // increment the number of evaluated parameter combinations
+        ++num_evaluated_parameter_combinations;
+
+        // take the first (and only) output BDL pair
+        const auto& output_bdl_pair = output_bdl_pairs.front();
+
+        // initialize a BDL input iterator
+        bdl_input_iterator<Lyt> bii{layout, params.bdl_params};
+
+        sidb_simulation_parameters sim_params = params.sim_params;
+        set_x_dimension_value(sim_params, param_point.x);
+        set_y_dimension_value(sim_params, param_point.y);
+
+        // for each input combination
+        for (auto i = 0u; i < truth_table.num_bits(); ++i, ++bii)
+        {
+            // the expected output of the layout is the i-th bit of the truth table
+            const auto expected_output = kitty::get_bit(truth_table, i);
+
+            ++num_simulator_invocations;
+
+            if (can_positive_charges_occur(*bii, sim_params))
+            {
+                return non_operational();
+            }
+
+            sidb_simulation_result<Lyt> sim_result{};
+
+            if (params.sim_engine == sidb_simulation_engine::EXGS)
+            {
+                // perform an exhaustive ground state simulation
+                sim_result = exhaustive_ground_state_simulation(*bii, sim_params);
+            }
+            else if (params.sim_engine == sidb_simulation_engine::QUICKSIM)
+            {
+                // perform a heuristic simulation
+                const quicksim_params qs_params{sim_params, 500, 0.6};
+                sim_result = quicksim(*bii, qs_params);
+            }
+            else if (params.sim_engine == sidb_simulation_engine::QUICKEXACT)
+            {
+                // perform fast exact ground state simulation
+                const quickexact_params<Lyt> qe_params{sim_params};
+                sim_result = quickexact<Lyt>(*bii, qe_params);
+            }
+            else
+            {
+                assert(false && "unsupported simulation engine");
+            }
+
+            // if no physically-valid charge distributions were found, the layout is non-operational
+            if (sim_result.charge_distributions.empty())
+            {
+                return non_operational();
+            }
+
+            // if the ground state is degenerate, the layout is non-operational
+            if (const auto energy_distr = energy_distribution(sim_result.charge_distributions);
+                energy_distr.count(0.0) > 0)
```

Comment:
@marcelwa This checks if the key ``0.0`` exists, right? If so, this would be wrong, because the ground state energy is different and only zero if all SiDBs are neutrally charged.

Reply:
Thanks, should be fixed now :+1: